test/Microsoft.AspNet.Mvc.Core.Test/ViewComponentTests.cs

```diff
@@ -180,7 +164,7 @@ public void ViewComponent_ViewContext_ViewData_ReturnsDefaultInstanceIfNull()
             Assert.Empty(viewComponent.ViewContext.ViewData);
             Assert.NotNull(viewComponent.ViewData);
             Assert.Empty(viewComponent.ViewData);
-            Assert.NotSame(viewComponent.ViewData, viewComponent.ViewContext.ViewData);
+            Assert.Same(viewComponent.ViewData, viewComponent.ViewContext.ViewData);
```

Comment:
We need to discuss this. The "defensive copy" causes a user experience different from that in other contexts e.g. within a controller action.